File: backend/BulkApi/Models/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace BulkApi.Models
{
    public class Product
    {
        public int ProductId { get; set; }
        public string Name { get; set; }
        public string Category { get; set; }
        public string Description { get; set; }
        public double OriginalPrice { get; set; }

        //FK ProducerId
        public string ProducerId { get; set; }
        public Producer Producer { get; set; }

        public List<DiscountScheme> DiscountSchemes { get; set; }
    }
}

File: frontend/src/models/Product.ts
import { DiscountScheme } from "./DiscountScheme";

export class Product {
    productId: number = 0;
    name: string = "";
    category: string = "";
    description: string = "";
    originalPrice: number = 0;
    producerId: number = 0;

    discountSchemes: DiscountScheme[] = [];

}